server-base.R

```diff
@@ -68,11 +71,39 @@ shinyServer(function(input, output, session){
     to_plot$r_quiet@data <<- cbind(to_plot$r_quiet@data[!(names(to_plot$r_quiet) %in% names(to_plot$l))], to_plot$l@data)
     # Add rqincr column to the quiet data
     to_plot$r_quiet@data$rqincr <<- to_plot$r_quiet@data$length / to_plot$r_fast@data$length
+    region$repopulate_region <<- F
+    #isolate(region$replot <- !region$replot)
+  })
+
+  region <- reactiveValues(current = starting_city, data_dir = file.path(data_dir_root, starting_city), repopulate_region = F) # replot = F,
 
-    isolate(region$replot <- !region$replot)
+  observe({
+    # If a region does not have an 'all-trips'directory, disable the dropdown menu
+    if (!dir.exists(file.path(data_dir_root, starting_city, 'all-trips'))){
```

Comment:
Your doing this check in a few places.  Maybe you could store it as a reactive value on `region` here,
e.g.
```R
region$all_trips <- dir.exists(file.path(data_dir_root, starting_city, 'all-trips')
```
then check for that in other places?